### DISCUSSION TITLE
Set molecule in Ketcher exactly when it is open

### DISCUSSION THREAD

**Discussion by EugenReentovich** [epam/ketcher#1049] [General]
Hello everyone!

I already created a similar issue https://github.com/epam/ketcher/issues/601 , but didn't get what I expected. I don' want to make any changes in Ketcher, so, I need some API with which I can interact from the outside. E.g., I could define in Ketcher's parent window some 'onKetcherLoad' function. The Ketcher could read the function and, if defined, execute it on load.
And yes, this problem is easy to solve with timers, but I would like to avoid it.

So, can Ketcher read functions from window.top? Maybe you have another ideas to solve this.

**Comment by AndreiMazol** [comment]
Dear @EugenReentovich ,

sorry for delay I missed your message. To set molecule exactly  after initialization of canvas you  can you onInit property on Editor React component:
`<Editor
        errorHandler={(message: string) => {
          setHasError(true)
          setErrorMessage(message.toString())
        }}
        buttons={hiddenButtonsConfig}
        staticResourcesUrl={process.env.PUBLIC_URL!}
        structServiceProvider={structServiceProvider}
        onInit={(ketcher: Ketcher) => {
         ketcher.setMolecule({your string here})
        }}
      />`

Best regards,
Andrei
